proposals/lip-0058.md

```diff
@@ -626,26 +674,50 @@ The function allows to obtain the next larger key in the BFT Parameters substore
 
 For a height `h` as input, the function returns the smallest height `h1 > h` with an entry in the BFT Parameters substore with key `h1`, if such height exists. Otherwise, it fails.
 
+
+#### getSlotNumber
+
+This function returns the slot number corresponding to the input timestamp.
+
+```python
+def getSlotNumber(timestamp):
+    if timestamp < genesisData.timestamp:
+        raise Exception('Input timestamp cannot be smaller than genesis block timestamp.')
+    elapsedTime = timestamp - genesisData.timestamp
+    return floor(elapsedTime / BLOCK_TIME)
+```
+
+#### getValidator
+
+For an address and height as input, the function returns the validator information of that validator at the given height. The function raises an exception if the validator is not in the validator set at that height.
+
+```python
+def getValidator(address, height):
+    bftParams = getBFTParameters(height)
+    for val in bft.validators:
+        if val.address = address:
+            return val
+    raise Exception()
+```
+
 #### setBFTParameters
 
-This function allows to set the thresholds, validators and associated BFT weights to be used from the next height onward and updates the BFT module store accordingly.
+This function allows to set the thresholds, validators and associated BFT weights to be used from the next height onward and updates the BFT module store accordingly. The function should be called with the parameters forwarded to the application layer from the state machine every time the function [setValidatorParams][lip-0044#setValidatorParams] of the Validators module is called.
```

Comment:
Updated to

> The function is called in the after application processing stage of the genesis block processing and general block processing as defined below.

In this LIP I use the more general academic terminology (state machine, replication layer) instead of the concrete Lisk SDK architecture terminology (application domain, engine domain). Not sure if this is the best approach, I will raise this point on Slack.

Reply:
As concluded on Slack, I now only use the terms application domain and consensus domain as introduced in LIP 69.